core/trino-main/src/main/java/io/trino/cost/FilterProjectAggregationStatsRule.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.cost;
+
+import io.trino.Session;
+import io.trino.matching.Pattern;
+import io.trino.sql.planner.TypeProvider;
+import io.trino.sql.planner.iterative.GroupReference;
+import io.trino.sql.planner.iterative.Lookup;
+import io.trino.sql.planner.plan.AggregationNode;
+import io.trino.sql.planner.plan.FilterNode;
+import io.trino.sql.planner.plan.PlanNode;
+import io.trino.sql.planner.plan.ProjectNode;
+
+import java.util.Optional;
+
+import static com.google.common.collect.MoreCollectors.onlyElement;
+import static io.trino.SystemSessionProperties.isUnestimatedFilterCostApproximationEnabled;
+import static io.trino.cost.FilterStatsCalculator.UNKNOWN_FILTER_COEFFICIENT;
+import static io.trino.sql.planner.plan.Patterns.filter;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * AggregationStatsRule does not have sufficient information to provide accurate estimates on aggregated symbols.
+ * This rule finds unestimated filters on top of aggregations whose output row count was estimated and returns
+ * estimated row count of (0.9 * aggregation output row count) for the filter.
+ */
+public class FilterProjectAggregationStatsRule
+        extends SimpleStatsRule<FilterNode>
+{
+    private static final Pattern<FilterNode> PATTERN = filter();
+
+    private final FilterStatsCalculator filterStatsCalculator;
+
+    public FilterProjectAggregationStatsRule(StatsNormalizer normalizer, FilterStatsCalculator filterStatsCalculator)
+    {
+        super(normalizer);
+        this.filterStatsCalculator = requireNonNull(filterStatsCalculator, "filterStatsCalculator cannot be null");
+    }
+
+    @Override
+    public Pattern<FilterNode> getPattern()
+    {
+        return PATTERN;
+    }
+
+    @Override
+    protected Optional<PlanNodeStatsEstimate> doCalculate(FilterNode node, StatsProvider sourceStats, Lookup lookup, Session session, TypeProvider types)
+    {
+        if (!isUnestimatedFilterCostApproximationEnabled(session)) {
+            return Optional.empty();
+        }
+        PlanNode nodeSource = resolveGroup(lookup, node.getSource());
+        AggregationNode aggregationNode;
+        if (nodeSource instanceof ProjectNode) {
+            ProjectNode projectNode = (ProjectNode) nodeSource;
+            if (!projectNode.isIdentity()) {
```

Comment:
why only identify?

Reply:
I was trying to match, as closely as I could, for the cases where we don't know the estimate due to lack of aggregate symbol stats. For non-identity projections I wasn't sure if lack of estimate could be due to an unestimated projection. We also don't have a non-indentity projection case in TPC benchmarks.